jobrunner/job_executor.py

```diff
@@ -0,0 +1,171 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Mapping, List, Tuple, Optional
+
+from jobrunner.models import State, StatusCode
+from jobrunner import config
+
+
+class Privacy(Enum):
+    HIGH = "high"
+    MEDIUM = "medium"
+
+
+@dataclass
+class Study:
+    git_repo_url: str
+    commit: str
+
+
+@dataclass
+class JobDefinition:
+    id: str  # a unique identifier for the job
+    study: Study  # the study defining the action for this job
+    workspace: str  # the workspace to run the job in
+    action: str  # the name of the action that the job is running
+    image: str  # the Docker image to run
+    args: List[str]  # the arguments to pass to the Docker container
+    env: Mapping[str, str]  # the environment variables to set for the Docker container
+    inputs: List[str]  # the files that the job requires
+    output_spec: Mapping[
+        str, str
+    ]  # the files that the job should produce (globs mapped to privacy levels)
+    allow_database_access: bool  # whether this job should have access to the database
+
+
+@dataclass
+class JobResults:
+    state: State
+    status_code: Optional[StatusCode]
+    status_message: str
+    outputs: Mapping[str, str]
+    exit_code: int
+    image_id: str
```

Comment:
Note: I added these to be able to write the manifest via API. They may not survive, perhaps.

I think `exit_code` would still be good debugging info, but image_id will hopefully be determined a priori rather than post at some point.